mkdocs_redirects/plugin.py

```diff
@@ -65,31 +66,35 @@ def get_relative_html_path(old_page, new_page, use_directory_urls):
     if use_directory_urls:
         relative_path = relative_path + '/'
 
-    return relative_path
+    return relative_path + new_hash_fragment
 
 
 def get_html_path(path, use_directory_urls):
     """ Return the HTML file path for a given markdown file """
-    parent, filename = posixpath.split(path)
+    path_hash_free, hash_fragment = _split_hash_fragment(path)
+    parent, filename = posixpath.split(path_hash_free)
     name_orig = posixpath.splitext(filename)[0]
 
     # Both `index.md` and `README.md` files are normalized to `index.html` during build
     name = 'index' if name_orig in ('index', 'README') else name_orig
 
     # Directory URLs require some different logic. This mirrors mkdocs' internal logic.
     if use_directory_urls:
+        index_name = 'index.html' + hash_fragment
 
         # If it's name is `index`, then that means it's the "homepage" of a directory, so should get placed in that dir
         if name == 'index':
-            return posixpath.join(parent, 'index.html')
+            html_path_segments = [index_name]
 
         # Otherwise, it's a file within that folder, so it should go in its own directory to resolve properly
         else:
-            return posixpath.join(parent, name, 'index.html')
+            html_path_segments = [name, index_name]
 
     # Just use the original name if Directory URLs aren't used
     else:
-        return posixpath.join(parent, (name + '.html'))
+        html_path_segments = [name + '.html' + hash_fragment]
+
+    return posixpath.join(parent, *html_path_segments)
```

Comment:
This can be done much simpler. Assuming all old code above, just replace `return` with `new_path = `.
And then
```suggestion
    return path + hash_fragment
```